### PR TITLE
Feat(eos_designs): Support routing protocol option on l3_edge p2p_links

### LINKED ISSUE #3493: Feat(eos_designs): Support BGP ipv4 unicast peering on l3_edge.p2p_links within a RFC5549 enabled fabric
### Enhancement summary

Implement an additional key in `l3_edge.p2p_profiles` and `l3_edge.p2p_links` to explicitly ignore `underlay_rfc5549: true`.

### Which component of AVD is impacted

eos_designs

### Use case example

If a fabric is set to use RFC5549 for underlay BGP peering, the data model for l3_edge.p2p_links supports only RFC5549 BGP peering to devices outside of the fabric. But we cannot ensure that all 3rd party devices support RFC5549 or outside device's administrators commit on RFC5549. Therefor it would be very helpful to have a knob to instruct the role to render a classic IPv4 BGP peering based on the IP information provided within the p2p link.

### Describe the solution you would like

````yaml
underlay_routing_protocol: ebgp
underlay_rfc5549: true
l3_edge:
  p2p_links:
    - nodes:
        - borderleaf1
        - campus-core1
      subnet: 10.99.0.0/30
      include_in_underlay_protocol: true  # this is default, but helpful for the context
      ignore_rfc5549: true  # default false
      interfaces:
        - Ethernet1
        - Any
 ````

should render this BGP configuration on `borderleaf1`:

````code
router bgp 65101
   ...
   neighbor IPv4-UNDERLAY-NON-RFC5549-PEERS peer group
   neighbor IPv4-UNDERLAY-NON-RFC5549-PEERS password 7 <removed>
   neighbor IPv4-UNDERLAY-NON-RFC5549-PEERS send-community
   neighbor IPv4-UNDERLAY-NON-RFC5549-PEERS maximum-routes 12000
   ...
   neighbor 10.99.0.2 peer group IPv4-UNDERLAY-NON-RFC5549-PEERS
   neighbor 10.99.0.2 remote-as 65499
   ...
   address-family ipv4
      ...
      neighbor IPv4-UNDERLAY-NON-RFC5549-PEERS activate
````

### Describe alternatives you have considered

Using `custom_structured_config_router_bgp` and add peer group and peers manually.
But this leads to duplicate data within the inventory for IP addresses used in l3_edge.p2p_links and in custom_structured_config.

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kornoa**
## Change Summary

Implement an additional key in
- `l3_edge.p2p_links_profiles[].routing_protocol`
- `l3_edge.p2p_links[].routing_protocol`
- `core_interfaces.p2p_links_profiles[].routing_protocol` and
- `core_interfaces.p2p_links[].routing_protocol`

to overwrite underlay routing protocol used on the link. Currently only `ebgp` is implemented which enforces classic IPv4 BGP Peering in a RFC5549 enabled fabric.

## Related Issue(s)

Fixes #3493

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
- Added the new keys to the role schema
- Changed the structured config generation in corresponding underlay python module

## How to test
- Added a test case to molecule scenario `evpn_underlay_rfc5549_overlay_ebgp`

### Repository Checklist

- [ x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/evpn_underlay_rfc5549_overlay_ebgp/intended/structured_configs/DC1-BL1A.yml [hunk 1]
```diff
@@ -337,6 +342,16 @@ ethernet_interfaces:
...
+- name: Ethernet11
+  peer: outside-r1
+  peer_interface: other1
+  peer_type: other
+  description: P2P_LINK_TO_outside-r1_other1
+  type: routed
+  shutdown: false
+  mtu: 1500
+  ip_address: 10.23.23.1/30
+  ipv6_enable: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/evpn_underlay_rfc5549_overlay_ebgp/intended/structured_configs/DC1-BL1A.yml]: You have set ipv6_enable specifically to False. That should not even be necessary. Instead add your extra condition for `routing_protocol != "ebgp"` to the place where we generate ipv6 based on 5549. (It should still be possible to generate ipv6_enable if specifically set to true)

Comment (kornoa) [reply ansible_collections/arista/avd/molecule/evpn_underlay_rfc5549_overlay_ebgp/intended/structured_configs/DC1-BL1A.yml]: Got it. I'm on vacation this week. I'll commit the change to the PR next week.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/evpn_underlay_rfc5549_overlay_ebgp/intended/structured_configs/DC1-BL1A.yml]: I have put this in draft for now. Set as ready once you have updated. Thanks.

Comment (kornoa) [reply ansible_collections/arista/avd/molecule/evpn_underlay_rfc5549_overlay_ebgp/intended/structured_configs/DC1-BL1A.yml]: Done. Please review.

**Review by Shivani-gslab — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]

**Comment by ClausHolbechArista** [comment]
Approved. I will resolve the conflict once we have merged #3584.

**Pull request merged**